integration-cli/docker_cli_swarm_test.go

```diff
@@ -194,3 +194,19 @@ func (s *DockerSwarmSuite) TestSwarmNodeTaskListFilter(c *check.C) {
 	c.Assert(out, checker.Not(checker.Contains), name+".2")
 	c.Assert(out, checker.Not(checker.Contains), name+".3")
 }
+
+// Test case for #25375
+func (s *DockerSwarmSuite) TestSwarmPublishAdd(c *check.C) {
```

Comment:
Thanks @dnephin I added a couple of more unit tests. I temporarily leave the integration test alone, as some of the test case will need to rely on swarmkit. Let me know if we feel unit test is ok to cover the above mentioned scenarios and I will update the PR accordingly.